pkgs/build-support/shell-builder.nix

```diff
@@ -0,0 +1,45 @@
+{ lib, stdenv }:
+
+# A special kind of derivation that is only meant to be consumed by the
+# nix-shell.
+{
+  mergeInputs ? [], # a list of derivations whose inputs will be merged
+  buildInputs ? [],
+  nativeBuildInputs ? [],
+  propagatedBuildInputs ? [],
+  ...
```

Comment:
You are missing `propagatedNativeBuildInputs` here, and what is the use of `...`?

Reply:
fixed